batcher-bot/liquidity/src/bot.ts

```diff
@@ -0,0 +1,231 @@
+import { HubConnection } from "@microsoft/signalr";
+import {
+  contract_details,
+  liquidity_settings,
+  batch_provision,
+  token_pair,
+} from "./types";
+import { parse_deposit } from "./utils";
+import { submit_deposit, submit_redemption } from "./submitter";
+import { can_provision_always_on, can_provision_jit } from "./provision";
+import { TezosToolkit } from "@taquito/taquito";
+
+const redeem_on_cleared = (
+  batcher_contract: string,
+  msg: any,
+  tezos: TezosToolkit
+) => {
+  for (let i = 0; i < Object.keys(msg.data).length; i++) {
+    try {
+      const message = msg.data[i];
+      if (message.path == "batch_set.batches") {
+        const val = message.content.value;
+        const batch_number = val.batch_number;
+        const status = Object.keys(val.status)[0];
+        if (status == "cleared") {
+          console.info(`Batch ${batch_number} was cleared. Redeeming `);
+          submit_redemption(batcher_contract, tezos).then(() => {
+            console.info(`Redeemed holdings `);
+          });
+        }
+      }
+    } catch (error: any) {
+      console.info("Error parsing bigmap for redemption", error);
+      console.error(error);
+    }
+  }
+};
+const getPairName = (fromName: string, toName: string) => {
+  if (fromName > toName) {
+    return fromName + "/" + toName;
+  }
+
+  return toName + "/" + fromName;
+};
+
+const parse_status = (status:any) => {
+
+  try{
+    const status_keys = Object.keys(status)[0];
+   console.info("Status keys", status_keys);
+   console.info("Status", status);
+    if (status_keys.includes("cleared")){
+      return `cleared @ ${status.cleared.at}`;
+    }
+
+    if (status_keys.includes("closed")){
+      return `closed @ ${status.closed.closing_time}`;
+    }
```

Comment:
If you can, it would be better to have an `any` and rather use a union type (e.g `type1 | type2`)

Even if you refactor or not, rather than getting all the keys and check if it's inside with an include, you can do: `if ('cleared' in status)`

Reply:
Thanks, changed